8316178: Better diagnostic header for CodeBlobs

shipilev opened the pull request:
I realized that JMH perfasm misses some CodeBlobs, notably C1 RuntimeBlobs. While amending perfasm to capture those, I discovered the header printout for those blobs is not as convenient as for others. Notably, perfasm would like to know where the stub code ends.

```
# Good:
StubRoutines::sha512_implCompressMB [0x0000000117da2d80, 0x0000000117da3528] (1960 bytes)

# Bad:
Decoding RuntimeStub - _complete_monitor_locking_Java 0x0000000117e4fd10
```

After this change, we print this:

```
Decoding RuntimeStub - _complete_monitor_locking_Java 0x0000000114203110 [0x00000001142031c0, 0x0000000114203258] (152 bytes)
```

Note it still prints the (base) stub address. I think it is still useful for debugging. The brackets print the code size, because that is what the decoding block handles: the code.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8316178](https://bugs.openjdk.org/browse/JDK-8316178): Better diagnostic header for CodeBlobs (**Enhancement** - P4)


### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15708/head:pull/15708` \
`$ git checkout pull/15708`

Update a local copy of the PR: \
`$ git checkout pull/15708` \
`$ git pull https://git.openjdk.org/jdk.git pull/15708/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15708`

View PR using the GUI difftool: \
`$ git pr show -t 15708`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15708.diff">https://git.openjdk.org/jdk/pull/15708.diff</a>

</details>


### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15708#issuecomment-1717395203)

TobiHartmann reviewed (approved):
Looks good to me.

shipilev commented:
Thanks @TobiHartmann! I think I would need another review before I can integrate.

vnkozlov reviewed (approved):
Looks good.
One question: did  you change `INTPTR_FORMAT` to `PTR_FORMAT` because they are the same but shorter name?

shipilev commented:
> Looks good. One question: did you change `INTPTR_FORMAT` to `PTR_FORMAT` because they are the same but shorter name?

Muscle memory: when using `p2i` I use `PTR_FORMAT`, so when I rewrote `stub` to `p2i(stub)`, I also changed to `PTR_FORMAT` to match. Not really sure what is our agreed style here, though.

vnkozlov commented:
> > Looks good. One question: did you change `INTPTR_FORMAT` to `PTR_FORMAT` because they are the same but shorter name?
> 
> Muscle memory: when using `p2i` I use `PTR_FORMAT`, so when I rewrote `stub` to `p2i(stub)`, I also changed to `PTR_FORMAT` to match. Not really sure what is our agreed style here, though.

It looks like we use both evenly so I am fine with this change.

stefank commented:
> > > Looks good. One question: did you change `INTPTR_FORMAT` to `PTR_FORMAT` because they are the same but shorter name?
> > 
> > 
> > Muscle memory: when using `p2i` I use `PTR_FORMAT`, so when I rewrote `stub` to `p2i(stub)`, I also changed to `PTR_FORMAT` to match. Not really sure what is our agreed style here, though.
> 
> It looks like we use both evenly so I am fine with this change.

We have been pushing people towards using PTR_FORMAT instead of INTPTR_FORMAT when printing pointers.

shipilev commented:
Okay, thanks for reviews. I'll integrate and see to JMH perfasm changes. Once that is done, I am going to backport this to actively maintained JDK releases.

/integrate

shipilev commented:
/backport jdk21u

shipilev commented:
/backport jdk17u-dev
